modules/gdscript/language_server/gdscript_extend_parser.cpp

```diff
@@ -797,9 +822,24 @@ Dictionary ExtendGDScriptParser::dump_class_api(const GDScriptParser::ClassNode
 			case ClassNode::Member::SIGNAL: {
 				Dictionary api;
 				api["name"] = m.signal->identifier->name;
-				Array pars;
+				Dictionary pars;
 				for (int j = 0; j < m.signal->parameters.size(); j++) {
-					pars.append(String(m.signal->parameters[i]->identifier->name));
+					const GDScriptParser::ParameterNode &this_parameter = *m.signal->parameters[i];
+					// has signal_arg_type?
+					Variant::Type type = Variant::NIL;
+					if (this_parameter.datatype_specifier && this_parameter.datatype_specifier->type_chain.size() == 1) {
+						const Variant::Type builtin_type = GDScriptParser::get_builtin_type(this_parameter.datatype_specifier->type_chain[0]->name);
+						// is builtin ?
+						// todo: If in `GDScriptParser::parse_signal()` it can be assumed that the type is always a built-in type then the following test is not required.
```

Comment:
```suggestion
						// TODO: If in `GDScriptParser::parse_signal()` it can be assumed that the type is always a built-in type, then the following test is not required.
```